memcache/memcache.go

```diff
@@ -130,7 +131,30 @@ var (
 func New(server ...string) *Client {
 	ss := new(ServerList)
 	_ = ss.SetServers(server...)
-	return NewFromSelector(ss)
+	c := NewFromSelector(ss)
+	c.serverList = append(c.serverList, server...)
+
+	return c
+}
+
+// NewWithBackgroundReconnect returns the same thing as New(server ...string), but takes a context to initiate a bakground reconnect cron
+func NewWithBackgroundReconnect(ctx context.Context, server ...string) *Client {
+	c := New(server...)
+
+	// periodically ping the provided servers -- if there are timeouts, it will trigger a reconnect attempt
+	go func() {
+		ticker := time.NewTicker(10 * time.Minute)
```

Comment:
I think `ticker.Stop()` needs to be called at some point

Reply:
Probably good to make it ticker interval configurable

Reply:
Both done [428ba0a](https://github.com/grafana/gomemcache/pull/18/commits/428ba0a456d168730751c471bbede70b851031aa)